IOUtilTest.test_zipAndUnzipNestedFolder_then_contentsShouldBeSame

failed on _master_ PR builder

Failed on http://jenkins.hazelcast.com/job/jet-oss-pr-builder/1846/testReport/junit/com.hazelcast.jet.impl.util/IOUtilTest/test_zipAndUnzipNestedFolder_then_contentsShouldBeSame/

Stacktrace:
```
java.lang.AssertionError: expected:<7> but was:<6>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.test.HazelcastTestSupport.assertCollection(HazelcastTestSupport.java:180)
	at com.hazelcast.jet.impl.util.IOUtilTest.test(IOUtilTest.java:71)
	at com.hazelcast.jet.impl.util.IOUtilTest.test_zipAndUnzipNestedFolder_then_contentsShouldBeSame(IOUtilTest.java:48)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output:
```
Finished Running Test: test_zipAndUnzipNestedFoldersWithAnEmptySubFolder_then_contentsShouldBeSame in 0.411 seconds.
BuildInfo right after test_zipAndUnzipNestedFolder_then_contentsShouldBeSame(com.hazelcast.jet.impl.util.IOUtilTest): BuildInfo{version='4.0-SNAPSHOT', build='20200110', buildNumber=20200110, revision=bbf6a5b, enterprise=false, serializationVersion=1, jet=JetBuildInfo{version='4.0-SNAPSHOT', build='20200110', revision='ee03c4f'}}
Hiccups measured while running test 'test_zipAndUnzipNestedFolder_then_contentsShouldBeSame(com.hazelcast.jet.impl.util.IOUtilTest):'
13:00:25, accumulated pauses: 807 ms, max pause: 383 ms, pauses over 1000 ms: 0

```

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/util/IOUtilTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/util/IOUtilTest.java
@@ -18,7 +18,8 @@ package com.hazelcast.jet.impl.util;
 
 import com.hazelcast.jet.core.JetTestSupport;
 import com.hazelcast.jet.core.test.JetAssert;
-import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.Repeat;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
@@ -36,7 +37,7 @@ import java.util.stream.Collectors;
 import static com.hazelcast.jet.impl.util.IOUtil.packDirectoryIntoZip;
 import static com.hazelcast.jet.impl.util.IOUtil.unzip;
 
-@RunWith(HazelcastParallelClassRunner.class)
+@RunWith(HazelcastSerialClassRunner.class)
 public class IOUtilTest extends JetTestSupport {
 
     @Rule
@@ -51,8 +52,13 @@ public class IOUtilTest extends JetTestSupport {
     @Test
     public void test_zipAndUnzipNestedFoldersWithAnEmptySubFolder_then_contentsShouldBeSame() throws Exception {
         Path originalPath = Paths.get(this.getClass().getResource("/nested").toURI());
-        Files.createDirectory(originalPath.resolve(randomName())).toFile().deleteOnExit();
-        test(originalPath);
+        Path randomFile = originalPath.resolve(randomName());
+        Files.createDirectory(randomFile);
+        try {
+            test(originalPath);
+        } finally {
+            com.hazelcast.internal.nio.IOUtil.delete(randomFile);
+        }
     }
 
     public void test(Path originalPath) throws IOException {

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/util/IOUtilTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/util/IOUtilTest.java
@@ -19,7 +19,6 @@ package com.hazelcast.jet.impl.util;
 import com.hazelcast.jet.core.JetTestSupport;
 import com.hazelcast.jet.core.test.JetAssert;
 import com.hazelcast.test.HazelcastSerialClassRunner;
-import com.hazelcast.test.annotation.Repeat;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;

